### ISSUE TITLE
--print-dot is not under test

### ISSUE THREAD

**Issue by mr-c** [common-workflow-language/cwltool#315]
```
~/cwltool$ cwltool --debug --print-dot ~/common-workflow-language/v1.0/examples/nestedworkflows.cwl
/home/michael/cwltool/env/bin/cwltool 1.0.20170228095828
Resolved '/home/michael/common-workflow-language/v1.0/examples/nestedworkflows.cwl' to 'file:///home/michael/common-workflow-language/v1.0/examples/nestedworkflows.cwl'
I'm sorry, I couldn't load this CWL file.  The error was:
Traceback (most recent call last):
  File "/home/michael/cwltool/env/local/lib/python2.7/site-packages/cwltool/main.py", line 671, in main
    printdot(tool, document_loader.ctx, stdout)
  File "/home/michael/cwltool/env/local/lib/python2.7/site-packages/cwltool/cwlrdf.py", line 165, in printdot
    g = gather(wf, ctx)
  File "/home/michael/cwltool/env/local/lib/python2.7/site-packages/cwltool/cwlrdf.py", line 16, in gather
    tool.visit(visitor)
  File "/home/michael/cwltool/env/local/lib/python2.7/site-packages/cwltool/workflow.py", line 457, in visit
    op(self.tool)
  File "/home/michael/cwltool/env/local/lib/python2.7/site-packages/cwltool/cwlrdf.py", line 14, in visitor
    makerdf(t["id"], t, ctx, graph=g)
  File "/home/michael/cwltool/env/local/lib/python2.7/site-packages/schema_salad/jsonld_context.py", line 242, in makerdf
    g.parse(data=json.dumps(wf), format='json-ld', location=workflow)
  File "/home/michael/cwltool/env/local/lib/python2.7/site-packages/rdflib/graph.py", line 1030, in parse
    data=data, format=format)
  File "/home/michael/cwltool/env/local/lib/python2.7/site-packages/rdflib/parser.py", line 150, in create_input_source
    'exactly one of source, location, file or data must be given'
ValueError: exactly one of source, location, file or data must be given

```

**Comment by cnjr2** [comment]
I installed `cwltool` from source now, editing the `setup.py` for the version required and it works again.

I reverted back to `rdflib==4.2.1` instead of `rdflib==4.2.2`.

**Comment by mr-c** [comment]
Fixed by #318

**Comment by cnjr2** [comment]
I am still running into trouble with the `--print-dot` option.

The following workflow runs successfully:

```yaml
cwlVersion: v1.0
class: Workflow
inputs:
  message: string
outputs:
  message_file:
    type: File
    outputSource: step_one/output
steps:
  step_one:
    run:
      class: CommandLineTool
      baseCommand: echo
      stdout: output.txt
      inputs:
        message:
          type: string
          inputBinding:
            position: 1
      outputs:
        output:
          type: stdout
    in:
      message: message
    out: [output]
```

```
cwltool workflows/test.cwl --message=hello
/Users/user/anaconda/bin/cwltool 1.0.20170309164828
Resolved 'workflows/test.cwl' to 'file:///Users/user/workflows/test.cwl'
invalid field `job_order`, expected one of: 'message'
[job step_one] /private/var/folders/rk/jx0271ns4fq3ly5b82g4d91r0000gp/T/tmpI10PC2$ echo \
    hello > /private/var/folders/rk/jx0271ns4fq3ly5b82g4d91r0000gp/T/tmpI10PC2/output.txt
[job step_one] completed success
[step step_one] completed success
[workflow test.cwl] outdir is /var/folders/rk/jx0271ns4fq3ly5b82g4d91r0000gp/T/tmpnWxcuc
{
    "message_file": {
        "checksum": "sha1$f572d396fae9206628714fb2ce00f72e94f2258f",
        "basename": "output.txt",
        "location": "file:///Users/user/output.txt",
        "path": "/Users/user/output.txt",
        "class": "File",
        "size": 6
    }
}
Final process status is success
```

however `--print-dot` gets stuck.

```
cwltool --debug --print-dot workflows/test.cwl | dot -Tpdf -o test.pdf && open test.pdf
/Users/user/anaconda/bin/cwltool 1.0.20170309164828
Resolved 'workflows/test.cwl' to 'file:///Users/user/workflows/test.cwl'
I'm sorry, I couldn't load this CWL file.  The error was:
Traceback (most recent call last):
  File "/Users/user/anaconda/lib/python2.7/site-packages/cwltool/main.py", line 685, in main
    printdot(tool, document_loader.ctx, stdout)
  File "/Users/user/anaconda/lib/python2.7/site-packages/cwltool/cwlrdf.py", line 165, in printdot
    g = gather(wf, ctx)
  File "/Users/user/anaconda/lib/python2.7/site-packages/cwltool/cwlrdf.py", line 16, in gather
    tool.visit(visitor)
  File "/Users/user/anaconda/lib/python2.7/site-packages/cwltool/workflow.py", line 459, in visit
    s.visit(op)
  File "/Users/user/anaconda/lib/python2.7/site-packages/cwltool/workflow.py", line 597, in visit
    self.embedded_tool.visit(op)
  File "/Users/user/anaconda/lib/python2.7/site-packages/cwltool/process.py", line 620, in visit
    op(self.tool)
  File "/Users/user/anaconda/lib/python2.7/site-packages/cwltool/cwlrdf.py", line 14, in visitor
    makerdf(t["id"], t, ctx, graph=g)
  File "/Users/user/anaconda/lib/python2.7/site-packages/ruamel/yaml/comments.py", line 595, in __getitem__
    return ordereddict.__getitem__(self, key)
KeyError: 'id'
```

In this case I had to remove split the workflow into:

```yaml
cwlVersion: v1.0
class: Workflow

inputs:
  message: string

outputs:
  message_file:
    type: File
    outputSource: step_one/output

steps:
  step_one:
    run: echo.cwl
    in:
      message: message
    out: [output]
```

and an `echo.cwl`:

```yaml
cwlVersion: v1.0
class: CommandLineTool
baseCommand: echo
stdout: output.txt
inputs:
  message:
    type: string
    inputBinding:
      position: 1
outputs:
  output:
    type: stdout
```

For it the `--print-dot` to work.

```
cwltool --debug --print-dot workflows/test.cwl | dot -Tpdf -o test.pdf && open test.pdf
/Users/user/anaconda/bin/cwltool 1.0.20170309164828
Resolved 'workflows/test.cwl' to 'file:///Users/user/workflows/test.cwl'
```

[test.pdf](https://github.com/common-workflow-language/cwltool/files/838316/test.pdf)
